syntax = "proto3";

import "google/protobuf/empty.proto";

option go_package = "user_service/proto";

service UserService {
  rpc Register(UserRequest) returns (google.protobuf.Empty);
  rpc Login(UserRequest) returns (JWT);
  rpc Validate(JWT) returns (UserResponse);
  rpc FindById(UserIdRequest) returns (UserResponse);
  rpc FindByUsername(UsernameRequest) returns (UserResponse);
  rpc FindAll(google.protobuf.Empty) returns (UserListResponse);
  rpc DeleteById(UserIdRequest) returns (google.protobuf.Empty);
  rpc Status(google.protobuf.Empty) returns (StatusResponse);
}

message StatusResponse {
  string status = 1;
}

message UserRequest {
  string username = 1;
  string password = 2;
}

message UserIdRequest {
  string id = 1;
}

message UsernameRequest {
  string username = 1;
}

message UserResponse {
  string id = 1;
  string username = 2;
}

message JWT {
  string token = 1;
}

message UserListResponse {
  repeated UserResponse users = 1;
}